diff --git a/config/BuildSystem/config/packages/PTScotch.py b/config/BuildSystem/config/packages/PTScotch.py
index 3f857672d3..a3cf3bbb5b 100644
--- a/config/BuildSystem/config/packages/PTScotch.py
+++ b/config/BuildSystem/config/packages/PTScotch.py
@@ -4,7 +4,7 @@ class Configure(config.package.Package):
   def __init__(self, framework):
     config.package.Package.__init__(self, framework)
     self.version          = '6.0.9'
-    self.versionname      = 'SCOTCH_VERSION.SCOTCH_RELEASE.SCOTCH_PATCHLEVEL'
+    self.versionname      = 'SCOTCH_VERSION . SCOTCH_RELEASE . SCOTCH_PATCHLEVEL'
     self.gitcommit        = 'v'+self.version
     self.download         = ['git://https://gitlab.inria.fr/scotch/scotch.git',
                              'https://gitlab.inria.fr/scotch/scotch/-/archive/'+self.gitcommit+'/scotch-'+self.gitcommit+'.tar.gz',
diff --git a/config/BuildSystem/config/packages/PaStiX.py b/config/BuildSystem/config/packages/PaStiX.py
index a8f2fc3549..2584e90ce1 100644
--- a/config/BuildSystem/config/packages/PaStiX.py
+++ b/config/BuildSystem/config/packages/PaStiX.py
@@ -4,7 +4,7 @@ class Configure(config.package.Package):
   def __init__(self, framework):
     config.package.Package.__init__(self, framework)
     self.version          = '5.2.3'
-    self.versionname      = 'PASTIX_MAJOR_VERSION.PASTIX_MEDIUM_VERSION.PASTIX_MINOR_VERSION'
+    self.versionname      = 'PASTIX_MAJOR_VERSION . PASTIX_MEDIUM_VERSION . PASTIX_MINOR_VERSION'
     # 'https://gforge.inria.fr/frs/download.php/file/36212/pastix_'+self.version+'.tar.bz2',
     self.download         = ['http://ftp.mcs.anl.gov/pub/petsc/externalpackages/pastix_'+self.version+'.tar.bz2']
     self.liblist          = [['libpastix.a'],
diff --git a/config/BuildSystem/config/packages/SuiteSparse.py b/config/BuildSystem/config/packages/SuiteSparse.py
index afe38c7d55..c1feacb5ca 100644
--- a/config/BuildSystem/config/packages/SuiteSparse.py
+++ b/config/BuildSystem/config/packages/SuiteSparse.py
@@ -6,7 +6,7 @@ class Configure(config.package.Package):
     self.minversion        = '5.6.0'
     self.version           = '5.7.1'
     self.versioninclude    = 'SuiteSparse_config.h'
-    self.versionname       = 'SUITESPARSE_MAIN_VERSION.SUITESPARSE_SUB_VERSION.SUITESPARSE_SUBSUB_VERSION'
+    self.versionname       = 'SUITESPARSE_MAIN_VERSION . SUITESPARSE_SUB_VERSION . SUITESPARSE_SUBSUB_VERSION'
     self.gitcommit         = 'v'+self.version
     self.download          = ['git://https://github.com/DrTimothyAldenDavis/SuiteSparse','https://github.com/DrTimothyAldenDavis/SuiteSparse/archive/'+self.gitcommit+'.tar.gz']
     self.download_solaris  = ['https://github.com/DrTimothyAldenDavis/SuiteSparse/archive/v5.6.0.tar.gz']
diff --git a/config/BuildSystem/config/packages/SuperLU.py b/config/BuildSystem/config/packages/SuperLU.py
index 704fb58ddb..8b95c8447f 100644
--- a/config/BuildSystem/config/packages/SuperLU.py
+++ b/config/BuildSystem/config/packages/SuperLU.py
@@ -6,7 +6,7 @@ class Configure(config.package.CMakePackage):
     config.package.CMakePackage.__init__(self, framework)
     self.version          = '5.2.1'
     self.minversion       = '5.2.1' # bugs in 5.2.0 prevent it from functioning
-    self.versionname      = 'SUPERLU_MAJOR_VERSION.SUPERLU_MINOR_VERSION.SUPERLU_PATCH_VERSION'
+    self.versionname      = 'SUPERLU_MAJOR_VERSION . SUPERLU_MINOR_VERSION . SUPERLU_PATCH_VERSION'
     self.gitcommit        = 'a3d5233' # master mar-15-2020
     self.download         = ['git://https://github.com/xiaoyeli/superlu','https://github.com/xiaoyeli/superlu/archive/'+self.gitcommit+'.tar.gz']
     self.downloaddirnames  = ['SuperLU','superlu']
diff --git a/config/BuildSystem/config/packages/SuperLU_DIST.py b/config/BuildSystem/config/packages/SuperLU_DIST.py
index eb4cb74dc8..cef15f79e7 100644
--- a/config/BuildSystem/config/packages/SuperLU_DIST.py
+++ b/config/BuildSystem/config/packages/SuperLU_DIST.py
@@ -6,7 +6,7 @@ class Configure(config.package.CMakePackage):
     config.package.CMakePackage.__init__(self, framework)
     self.minversion       = '6.1.1'
     self.version          = '6.3.0'
-    self.versionname      = 'SUPERLU_DIST_MAJOR_VERSION.SUPERLU_DIST_MINOR_VERSION.SUPERLU_DIST_PATCH_VERSION'
+    self.versionname      = 'SUPERLU_DIST_MAJOR_VERSION . SUPERLU_DIST_MINOR_VERSION . SUPERLU_DIST_PATCH_VERSION'
     self.gitcommit        = 'v'+self.version
     self.download         = ['git://https://github.com/xiaoyeli/superlu_dist','https://github.com/xiaoyeli/superlu_dist/archive/'+self.gitcommit+'.tar.gz']
     self.downloaddirnames = ['SuperLU_DIST','superlu_dist']
diff --git a/config/BuildSystem/config/packages/metis.py b/config/BuildSystem/config/packages/metis.py
index c60ca3f138..7e49ebc3a1 100644
--- a/config/BuildSystem/config/packages/metis.py
+++ b/config/BuildSystem/config/packages/metis.py
@@ -3,7 +3,7 @@ import config.package
 class Configure(config.package.CMakePackage):
   def __init__(self, framework):
     config.package.CMakePackage.__init__(self, framework)
-    self.versionname       = 'METIS_VER_MAJOR.METIS_VER_MINOR.METIS_VER_SUBMINOR'
+    self.versionname       = 'METIS_VER_MAJOR . METIS_VER_MINOR . METIS_VER_SUBMINOR'
     self.gitcommit         = 'v5.1.0-p8'
     self.download          = ['git://https://bitbucket.org/petsc/pkg-metis.git','https://bitbucket.org/petsc/pkg-metis/get/'+self.gitcommit+'.tar.gz']
     self.downloaddirnames  = ['petsc-pkg-metis']
diff --git a/config/BuildSystem/config/packages/parmetis.py b/config/BuildSystem/config/packages/parmetis.py
index a48bb951e0..5f7a1c1b2a 100644
--- a/config/BuildSystem/config/packages/parmetis.py
+++ b/config/BuildSystem/config/packages/parmetis.py
@@ -4,7 +4,7 @@ class Configure(config.package.CMakePackage):
   def __init__(self, framework):
     config.package.CMakePackage.__init__(self, framework)
     self.version          = '4.0.3'
-    self.versionname      = 'PARMETIS_MAJOR_VERSION.PARMETIS_MINOR_VERSION.PARMETIS_SUBMINOR_VERSION'
+    self.versionname      = 'PARMETIS_MAJOR_VERSION . PARMETIS_MINOR_VERSION . PARMETIS_SUBMINOR_VERSION'
     self.gitcommit         = 'v'+self.version+'-p6'
     self.download          = ['git://https://bitbucket.org/petsc/pkg-parmetis.git','https://bitbucket.org/petsc/pkg-parmetis/get/'+self.gitcommit+'.tar.gz']
     self.functions         = ['ParMETIS_V3_PartKway']
diff --git a/config/BuildSystem/config/packages/strumpack.py b/config/BuildSystem/config/packages/strumpack.py
index 73bbb86a16..08006d12d4 100644
--- a/config/BuildSystem/config/packages/strumpack.py
+++ b/config/BuildSystem/config/packages/strumpack.py
@@ -5,7 +5,7 @@ class Configure(config.package.CMakePackage):
   def __init__(self, framework):
     config.package.CMakePackage.__init__(self, framework)
     self.version          = '3.1.1'
-    self.versionname      = 'STRUMPACK_VERSION_MAJOR.STRUMPACK_VERSION_MINOR.STRUMPACK_VERSION_PATCH'
+    self.versionname      = 'STRUMPACK_VERSION_MAJOR . STRUMPACK_VERSION_MINOR . STRUMPACK_VERSION_PATCH'
     self.versioninclude   = 'StrumpackConfig.hpp'
     self.gitcommit        = 'v'+self.version
     self.download         = ['git://https://github.com/pghysels/STRUMPACK','https://github.com/pghysels/STRUMPACK/archive/v'+self.version+'.tar.gz']
