diff --git a/config/BuildSystem/config/packages/hdf5.py b/config/BuildSystem/config/packages/hdf5.py
index 03ea5bdffb..309896a9f4 100644
--- a/config/BuildSystem/config/packages/hdf5.py
+++ b/config/BuildSystem/config/packages/hdf5.py
@@ -101,4 +101,5 @@ class Configure(config.package.GNUPackage):
       if output.find('foundbeast') > -1:
         if i.endswith('_H'): i = i[0:-2]
         self.addDefine('HDF5_HAVE_'+i, 1)
+    self.framework.addDefine('H5_BUILT_AS_DYNAMIC_LIB', 1)
 
