diff --git a/config/install.py b/config/install.py
index fc99b8398..34bf3e6c8 100755
--- a/config/install.py
+++ b/config/install.py
@@ -361,7 +361,8 @@ for dir in dirs:
     if not os.path.splitext(src)[1] == '.o':
       shutil.copy2(src, dst)
     if os.path.splitext(dst)[1] == '.'+self.arLibSuffix:
-      (result, output) = getstatusoutput(self.ranlib+' '+dst)
+      if not "msys" in self.arch:
+        (result, output) = getstatusoutput(self.ranlib+' '+dst)
     if os.path.splitext(dst)[1] == '.dylib' and os.path.isfile('/usr/bin/install_name_tool'):
       (result, output) = getstatusoutput('otool -D '+src)
       oldname = output[output.find("\n")+1:]
